deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/exception/GenericExceptionMapper.java

```diff
@@ -48,7 +48,11 @@ public Response toResponse(Throwable t) {
                 PrintWriter pw = new PrintWriter(sw);
                 t.printStackTrace(pw);
                 sb.append("\n").append(sw.toString());
-                return Response.serverError().entity(sb.toString()).build();
+                if (t.getMessage().contains("HTTP 4")) {
```

Comment:
```java
if (t instanceof ClientErrorException)
```

Reply:
I think we can handle it like dropwizard 
https://github.com/dropwizard/dropwizard/blob/0ae14874d8c39c314edc0bbe1a153aeb9db72320/dropwizard-jersey/src/main/java/io/dropwizard/jersey/errors/LoggingExceptionMapper.java#L38C1-L52C10

Reply:
updated and tested ok 

```
yaqinli@yaqinli-4JWJGH6 teletraan % curl -X POST --header 'Content-Type: application/json' --header 'Accept: application/json' --header 'Authorization: token eyJhbGciOiJSUzI1NiIsImtpZCI6Ijc2NDc3OTcwODAzODM0NDk4MSIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJGQTR2V2tiZEs4R0hxbFh6Und4ZnU2MlhKcTh0d0lOVjhMbGJGcnpTIiwiZXhwIjoxNjkwMjI3MjU3LCJncm91cHMiOlsiZW5naW5lZXJpbmciLCJpbmZyYS1zZWMiLCJkZWxpdmVyeS10ZWFtIiwianVweXRlci1hY2Nlc3MiLCJidXNpbmVzcy1zZW5zaXRpdmUtZHluYW1pYy1vbGQiLCJidXNpbmVzcy1zZW5zaXRpdmUtc3RhdGljIiwiZW5nLXByb2QiLCJqdXB5dGVyLWFjY2Vzcy10ZXN0IiwianVweXRlci1hY2Nlc3MtZHludGVtcCIsInRyYW5zcGFyZW5jeS1lbmctYWxsIiwiZ3BuLWJ1c2luZXNzLXNlbnNpdGl2ZS1keW5hbWljIiwiYnVzaW5lc3Mtc2Vuc2l0aXZlLWR5bmFtaWMiLCJncG4tYnVzaW5lc3Mtc2Vuc2l0aXZlIiwiY2RwIl0sImlhdCI6MTY4OTYyMjE1NywiaXNzIjoiaHR0cHM6Ly9hdXRoLnBpbmFkbWluLmNvbSIsImp0aSI6IjFlNzVjMWRjLTZkYjctNDU1My1hMTg3LTE2MzM0YTdkZTA5NCIsIm5iZiI6MTY4OTYyMjE1Nywic2NwIjpbInVzZXIiXSwic3ViIjoieWFxaW5saSJ9.L1-VqzdEjoI-ZnwBKdH2tdIhZCJF9mray21zvT2CSwI97Roz0LD1E6qiNj0n2sBJuzjSQ22GwsRyLdWrE6HtFYjNSqM4yby0BMBPO5OZmLxolKwInhpkgF30bK7OdivdwQIVdFf4ugA2tNLVEcuF7f_iMBj3ERIqbUqI1ha6Votab5GqwUnBw6RuSoRLndyEoay4T-FF-m-ELb-69fsziR3OBij8YD8Tz3GMLcwOI8AbfWLejZH1mY_nUjgfE4SbqeVJfsQ33Cg658meaexGnuKuJkOTWVrjDeJ_xU9ceVN8gw-MCYzpJ85DIk38sy5JzVxvQPQsgTPwa2yU_oH0mw' -d 'alert_name=response+codes&triggered=True&triggered_date=1510165897.62&empty_data_untriggered_date=None&empty_data_triggered_date=None&empty_data_triggered=False&alert_id=-1&untriggered_date=None' 'https://teletraan-dev1.pinadmin.com/v1/envs/deploy-open-sentinel-dev1/test-anhn/alerts?actionWindow=7200&actions=markbadbuild' --include
HTTP/1.1 400 Bad Request
date: Tue, 25 Jul 2023 23:13:12 GMT
content-length: 5860
x-envoy-upstream-service-time: 3
x-content-type-options: nosniff
strict-transport-security: max-age=631138519
x-frame-options: DENY
x-xss-protection: 1; mode=block
pragma: no-cache
cache-control: no-cache, no-store, must-revalidate
expires: Thu, 01 Jan 1970 00:00:00 GMT
referrer-policy: strict-origin-when-cross-origin
content-security-policy-report-only: script-src 'self'; frame-src 'self'; base-uri 'self'; report-uri https://www.pinterest.com/_/_/csp_report/?pinadmin=1&reportonly
envoy-instance: iwe
server: envoy


Message: HTTP 415 Unsupported Media Type
javax.ws.rs.NotSupportedException: HTTP 415 Unsupported Media Type
	at org.glassfish.jersey.server.internal.routing.MethodSelectingRouter.getMethodRouter(MethodSelectingRouter.java:478)
	at org.glassfish.jersey.server.internal.routing.MethodSelectingRouter.access$000(MethodSelectingRouter.java:94)
	at org.glassfish.jersey.server.internal.routing.MethodSelectingRouter$4.apply(MethodSelectingRouter.java:779)
	at org.glassfish.jersey.server.internal.routing.MethodSelectingRouter.apply(MethodSelectingRouter.java:371)
	at org.glassfish.jersey.server.internal.routing.RoutingStage._apply(RoutingStage.java:109)
	at org.glassfish.jersey.server.internal.routing.RoutingStage._apply(RoutingStage.java:112)
	at org.glassfish.jersey.server.internal.routing.RoutingStage._apply(RoutingStage.java:112)
	at org.glassfish.jersey.server.internal.routing.RoutingStage.apply(RoutingStage.java:92)
	at org.glassfish.jersey.server.internal.routing.RoutingStage.apply(RoutingStage.java:61)
	at org.glassfish.jersey.process.internal.Stages.process(Stages.java:197)
	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:318)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154)
	at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:473)
	at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:427)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:388)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:341)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:228)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:791)
	at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1626)
	at io.dropwizard.servlets.ThreadNameFilter.doFilter(ThreadNameFilter.java:35)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)
	at io.dropwizard.jersey.filter.AllowedMethodsFilter.handle(AllowedMethodsFilter.java:45)
	at io.dropwizard.jersey.filter.AllowedMethodsFilter.doFilter(AllowedMethodsFilter.java:39)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)
	at org.eclipse.jetty.servlets.CrossOriginFilter.handle(CrossOriginFilter.java:319)
	at org.eclipse.jetty.servlets.CrossOriginFilter.doFilter(CrossOriginFilter.java:273)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:548)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1435)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:501)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1350)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at com.codahale.metrics.jetty9.InstrumentedHandler.handle(InstrumentedHandler.java:313)
	at io.dropwizard.jetty.RoutingHandler.handle(RoutingHandler.java:52)
	at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:716)
	at io.dropwizard.jetty.BiDiGzipHandler.handle(BiDiGzipHandler.java:67)
	at org.eclipse.jetty.server.handler.RequestLogHandler.handle(RequestLogHandler.java:54)
	at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:179)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:516)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:388)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:633)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:380)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:773)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:905)
	at java.lang.Thread.run(Thread.java:750)
yaqinli@yaqinli-4JWJGH6 teletraan % 
```

Reply:
Accepted your suggestion and it is better:

```
yaqinli@yaqinli-4JWJGH6 teletraan % curl -X POST --header 'Content-Type: application/json' --header 'Accept: application/json' --header 'Authorization: token eyJhbGciOiJSUzI1NiIsImtpZCI6Ijc2NDc3OTcwODAzODM0NDk4MSIsInR5cCI6IkpXVCJ9.eyJhdWQiOiJGQTR2V2tiZEs4R0hxbFh6Und4ZnU2MlhKcTh0d0lOVjhMbGJGcnpTIiwiZXhwIjoxNjkwMjI3MjU3LCJncm91cHMiOlsiZW5naW5lZXJpbmciLCJpbmZyYS1zZWMiLCJkZWxpdmVyeS10ZWFtIiwianVweXRlci1hY2Nlc3MiLCJidXNpbmVzcy1zZW5zaXRpdmUtZHluYW1pYy1vbGQiLCJidXNpbmVzcy1zZW5zaXRpdmUtc3RhdGljIiwiZW5nLXByb2QiLCJqdXB5dGVyLWFjY2Vzcy10ZXN0IiwianVweXRlci1hY2Nlc3MtZHludGVtcCIsInRyYW5zcGFyZW5jeS1lbmctYWxsIiwiZ3BuLWJ1c2luZXNzLXNlbnNpdGl2ZS1keW5hbWljIiwiYnVzaW5lc3Mtc2Vuc2l0aXZlLWR5bmFtaWMiLCJncG4tYnVzaW5lc3Mtc2Vuc2l0aXZlIiwiY2RwIl0sImlhdCI6MTY4OTYyMjE1NywiaXNzIjoiaHR0cHM6Ly9hdXRoLnBpbmFkbWluLmNvbSIsImp0aSI6IjFlNzVjMWRjLTZkYjctNDU1My1hMTg3LTE2MzM0YTdkZTA5NCIsIm5iZiI6MTY4OTYyMjE1Nywic2NwIjpbInVzZXIiXSwic3ViIjoieWFxaW5saSJ9.L1-VqzdEjoI-ZnwBKdH2tdIhZCJF9mray21zvT2CSwI97Roz0LD1E6qiNj0n2sBJuzjSQ22GwsRyLdWrE6HtFYjNSqM4yby0BMBPO5OZmLxolKwInhpkgF30bK7OdivdwQIVdFf4ugA2tNLVEcuF7f_iMBj3ERIqbUqI1ha6Votab5GqwUnBw6RuSoRLndyEoay4T-FF-m-ELb-69fsziR3OBij8YD8Tz3GMLcwOI8AbfWLejZH1mY_nUjgfE4SbqeVJfsQ33Cg658meaexGnuKuJkOTWVrjDeJ_xU9ceVN8gw-MCYzpJ85DIk38sy5JzVxvQPQsgTPwa2yU_oH0mw' -d 'alert_name=response+codes&triggered=True&triggered_date=1510165897.62&empty_data_untriggered_date=None&empty_data_triggered_date=None&empty_data_triggered=False&alert_id=-1&untriggered_date=None' 'https://teletraan-dev1.pinadmin.com/v1/envs/deploy-open-sentinel-dev1/test-anhn/alerts?actionWindow=7200&actions=markbadbuild' --include
HTTP/1.1 415 Unsupported Media Type
date: Tue, 25 Jul 2023 23:42:45 GMT
content-length: 5860
x-envoy-upstream-service-time: 41
x-content-type-options: nosniff
strict-transport-security: max-age=631138519
x-frame-options: DENY
x-xss-protection: 1; mode=block
pragma: no-cache
cache-control: no-cache, no-store, must-revalidate
expires: Thu, 01 Jan 1970 00:00:00 GMT
referrer-policy: strict-origin-when-cross-origin
content-security-policy-report-only: script-src 'self'; frame-src 'self'; base-uri 'self'; report-uri https://www.pinterest.com/_/_/csp_report/?pinadmin=1&reportonly
envoy-instance: iwe
server: envoy


Message: HTTP 415 Unsupported Media Type
javax.ws.rs.NotSupportedException: HTTP 415 Unsupported Media Type
	at org.glassfish.jersey.server.internal.routing.MethodSelectingRouter.getMethodRouter(MethodSelectingRouter.java:478)
	at org.glassfish.jersey.server.internal.routing.MethodSelectingRouter.access$000(MethodSelectingRouter.java:94)
	at org.glassfish.jersey.server.internal.routing.MethodSelectingRouter$4.apply(MethodSelectingRouter.java:779)
	at org.glassfish.jersey.server.internal.routing.MethodSelectingRouter.apply(MethodSelectingRouter.java:371)
	at org.glassfish.jersey.server.internal.routing.RoutingStage._apply(RoutingStage.java:109)
	at org.glassfish.jersey.server.internal.routing.RoutingStage._apply(RoutingStage.java:112)
	at org.glassfish.jersey.server.internal.routing.RoutingStage._apply(RoutingStage.java:112)
	at org.glassfish.jersey.server.internal.routing.RoutingStage.apply(RoutingStage.java:92)
	at org.glassfish.jersey.server.internal.routing.RoutingStage.apply(RoutingStage.java:61)
	at org.glassfish.jersey.process.internal.Stages.process(Stages.java:197)
	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:318)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154)
	at org.glassfish.jersey.servlet.WebComponent.serviceImpl(WebComponent.java:473)
	at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:427)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:388)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:341)
	at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:228)
	at org.eclipse.jetty.servlet.ServletHolder.handle(ServletHolder.java:791)
	at org.eclipse.jetty.servlet.ServletHandler$ChainEnd.doFilter(ServletHandler.java:1626)
	at io.dropwizard.servlets.ThreadNameFilter.doFilter(ThreadNameFilter.java:35)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)
	at io.dropwizard.jersey.filter.AllowedMethodsFilter.handle(AllowedMethodsFilter.java:45)
	at io.dropwizard.jersey.filter.AllowedMethodsFilter.doFilter(AllowedMethodsFilter.java:39)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)
	at org.eclipse.jetty.servlets.CrossOriginFilter.handle(CrossOriginFilter.java:319)
	at org.eclipse.jetty.servlets.CrossOriginFilter.doFilter(CrossOriginFilter.java:273)
	at org.eclipse.jetty.servlet.FilterHolder.doFilter(FilterHolder.java:193)
	at org.eclipse.jetty.servlet.ServletHandler$Chain.doFilter(ServletHandler.java:1601)
	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:548)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1435)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:501)
	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1350)
	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at com.codahale.metrics.jetty9.InstrumentedHandler.handle(InstrumentedHandler.java:313)
	at io.dropwizard.jetty.RoutingHandler.handle(RoutingHandler.java:52)
	at org.eclipse.jetty.server.handler.gzip.GzipHandler.handle(GzipHandler.java:716)
	at io.dropwizard.jetty.BiDiGzipHandler.handle(BiDiGzipHandler.java:67)
	at org.eclipse.jetty.server.handler.RequestLogHandler.handle(RequestLogHandler.java:54)
	at org.eclipse.jetty.server.handler.StatisticsHandler.handle(StatisticsHandler.java:179)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
	at org.eclipse.jetty.server.Server.handle(Server.java:516)
	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:388)
	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:633)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:380)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273)
	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:105)
	at org.eclipse.jetty.io.ChannelEndPoint$1.run(ChannelEndPoint.java:104)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:773)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:905)
	at java.lang.Thread.run(Thread.java:750)
yaqinli@yaqinli-4JWJGH6 teletraan % 
```